shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1763,8 +1763,17 @@ void WebContents::OnGetDefaultPrinter(
     return;
   }
 
-  base::string16 printer_name =
-      device_name.empty() ? default_printer : device_name;
+  base::string16 printer_name;
```

Comment:
Can you remove https://github.com/electron/electron/blob/master/shell/browser/api/electron_api_web_contents.cc#L1848-L1856 and consolidate this block too,

```
base::string16 printer_name =
      device_name.empty() ? default_printer : device_name;
// If there are no valid printers available on the network, bail.
if (printer_name.empty() || !IsDeviceNameValid(printer_name)) {
   std::move(print_callback).Run(false, "no valid printers available");
   return;
 }
```

The current condition seems to omit device_name always.